### PR TITLE
CompareExcessVsMissing-method

### PR DESCRIPTION
**Pull request by RasmusBroborg**

### CODE REVIEW

File: Source_Projects/FdvsFileInformation/ValidationContext.cs [hunk 1]
```diff
@@ -31,26 +34,91 @@ private void CheckFileSizeForCorruptFile()
...
+    {
+      var unexpectedFiles = deliveryFiles.Where(f => f.FileInfoFlag == FdvsFileInfoFlag.Unexpected);
+      var missingFiles = crossReferenceFiles.Where(f => f.FileInfoFlag == FdvsFileInfoFlag.Missing);
+
+      foreach (var unexpectedFile in unexpectedFiles)
+      {
+        foreach (var missingFile in missingFiles)
+        {
+          //Check similarity excluding file extension:
+          var unexpectedFileWithoutFileExtension = unexpectedFile.FileName - unexpectedFile.FileExtension;
```
Comment (JohanSaisa) [inline Source_Projects/FdvsFileInformation/ValidationContext.cs:52]: string subtraction - new meta

File: Source_Projects/FdvsFileInformation/ValidationContext.cs [hunk 2]
```diff
@@ -31,26 +34,91 @@ private void CheckFileSizeForCorruptFile()
...
+          }
+        }
+      }
+    }
+    
+    /// <summary>
+    /// Calculates the difference between two string values. Lower value means closer similarity. 
+    /// 0 means that both strings are identical.
+    /// </summary>
+    /// <param name="str1"></param>
```
Comment (JohanSaisa) [inline Source_Projects/FdvsFileInformation/ValidationContext.cs:69]: follow same convention str vs string

File: Source_Projects/FdvsFileInformation/ValidationContext.cs [hunk 3]
```diff
@@ -31,26 +34,91 @@ private void CheckFileSizeForCorruptFile()
...
+    /// <param name="string2"></param>
+    /// <returns>The number of characters which differ between two string values.</returns>
+    private int LevenshteinDistance(string str1, string string2)
+    {
+      if (string.IsNullOrEmpty(str1))
+      {
+        if (string.IsNullOrEmpty(string2))
+          return 0;
```
Comment (JohanSaisa) [inline Source_Projects/FdvsFileInformation/ValidationContext.cs:77]: same row or brackets { }

File: Source_Projects/FdvsFileInformation/ValidationContext.cs [hunk 4]
```diff
@@ -31,26 +34,91 @@ private void CheckFileSizeForCorruptFile()
...
+      {
+        return str1.Length;
+      }
+
+      int str1Len = str1.Length;
+      int str2Len = string2.Length;
+      int[,] matrix = new int[str1Len + 1, str2Len + 1];
+
+      for (int i = 0; i <= str1Len; matrix[i, 0] = i++) ;
+      for (int j = 1; j <= str2Len; matrix[0, j] = j++) ;
```
Comment (JohanSaisa) [inline Source_Projects/FdvsFileInformation/ValidationContext.cs:91]: Leftover code?

File: Source_Projects/FdvsFileInformation/ValidationContext.cs [hunk 5]
```diff
@@ -31,26 +34,91 @@ private void CheckFileSizeForCorruptFile()
...
+          //Check similarity excluding file extension:
+          var unexpectedFileWithoutFileExtension = unexpectedFile.FileName - unexpectedFile.FileExtension;
+          var missingFileWithoutFileExtension = missingFile.FileName - missingFile.FileExtension;
+          var levensteinDistance = LevenshteinDistance(unexpectedFileWithoutFileExtension, missingFileWithoutFileExtension);
+
+          if (levensteinDistance <= 
+            filenameSimilarityPercentageThreshold * unexpectedFileWithoutFileExtension.Length || levensteinDistance <= 1)
+          {
+            missingFile.FileInfoFlagMessage = $"A potential match was found. Suggested file is located at {unexpectedFile.FilePath}.";
+          }
```
Comment (JohanSaisa) [inline Source_Projects/FdvsFileInformation/ValidationContext.cs:60]: Change flag on file

**Review by JohanSaisa — CHANGES_REQUESTED** [review]

File: Source_Projects/FdvsFileInformation/ValidationContext.cs (hunk 3 above)
Comment (RasmusBroborg) [reply Source_Projects/FdvsFileInformation/ValidationContext.cs:77]: Will need to return to this one at a later time.

File: Source_Projects/FdvsFileInformation/ValidationContext.cs (hunk 4 above)
Comment (RasmusBroborg) [reply Source_Projects/FdvsFileInformation/ValidationContext.cs:91]: The loop populates the matrix

**Review by JohanSaisa — APPROVED** [review]

**Pull request merged**